File: src/main/java/com/mpe85/grampa/event/PreParseEvent.java
package com.mpe85.grampa.event;

import com.mpe85.grampa.rule.RuleContext;

/**
 * Event posted before a parse run.
 * 
 * @author mpe85
 *
 * @param <T>
 *            the type of the stack elements
 */
public class PreParseEvent<T> extends RuleContextEvent<T> {
	
	public PreParseEvent(final RuleContext<T> context) {
		super(context);
	}
	
	
}

File: src/main/kotlin/com/mpe85/grampa/event/PreParseEvent.kt
package com.mpe85.grampa.event

import com.mpe85.grampa.rule.RuleContext

/**
 * Event posted before a parse run.
 *
 * @author mpe85
 * @param[T] The type of the stack elements
 * @param[context] The rule context in which the event was posted
 */
class PreParseEvent<T>(context: RuleContext<T>) : RuleContextEvent<T>(context)
